x/genutil/collect.go

```diff
@@ -0,0 +1,205 @@
+package genutil
+
+// DONTCOVER
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/crypto"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
```

Comment:
Although I agree that it makes sense to separate out this logic, I am not entirely convinced it should be a module, especially since it imports two other modules directly. Do we expect this to be used frequently by SDK chains other than `gaia`, in exactly this form?

If not, I wonder if a better structure might be to abstract out most of the common functionality into a library (subpackage of `types/` maybe) and call it from Gaia - then applications could import what they need.

Reply:
This seems to be a good pattern

Reply:
yup see my further comment, once the further abstraction is concluded, this should be able to be used by other blockchain projects with custom genesis account types and create validator messages. Thus it's good and ought to be a module. 

based on comments here: https://github.com/cosmos/cosmos-sdk/pull/4159#discussion_r282549364 I perceive it to be unwise to move have elements outside of `x/` depend on things _from_ `x/` (until the point that they are just using expected interfaces which is not yet the case) - so yeah I'm pretty fiercely certain this aint' belong in `types/` - but yeah see that other thread

Reply:
https://github.com/cosmos/cosmos-sdk/pull/4159#discussion_r282682194